scalding-core/src/main/scala/com/twitter/scalding/ArgHelp.scala

```diff
@@ -0,0 +1,106 @@
+package com.twitter.scalding
+
+sealed abstract class DescribedArg {
```

Comment:
Didn't give it too much thought.  I made the second one a trait so I can easily extend it for my unit tests.  I could make it an abstract class though.